modules/admin/proc-configuring-postgresql-with-tls-using-helm.adoc

```diff
@@ -0,0 +1,133 @@
+   <ca-certificate-key> <1>
+ postgres-key.key: |-
+   -----BEGIN CERTIFICATE-----
+   <tls-private-key> <2>
+ postgres-crt.pem: |-    
+   -----BEGIN CERTIFICATE-----
+   <tls-certificate-key> <3>
+   # ...
+----
+<1> Provide the CA certificate key.
+<2> Provide the TLS private key.
+<3> Provide the TLS certificate key.
+
+. Create a credential secret named `postgres-cred` to connect with the PostgreSQL instance:
++
+[source,yaml]
+----
+cat <<EOF | oc -n <your-namespace> create -f -
+apiVersion: v1
+kind: Secret
+metadata:
+ name: postgres-cred
+data: <1>
+  POSTGRES_PASSWORD: <password>
+  POSTGRES_PORT: "<db-port>"
+  POSTGRES_USER: <username>
+  POSTGRES_HOST: <db-host>
+  PGSSLMODE: <ssl-mode> # for TLS connection <2>
+EOF
+----
+<1> Provide credential data to connect with your PostgreSQL instance.
+<2> Provide the value based on the required link:https://www.postgresql.org/docs/15/libpq-connect.html#LIBPQ-CONNECT-SSLMODE[SSL mode].
+
+. Apply the `postgress-external-db-cluster-cert` secret to the namespace where you have deployed your {product-very-short} instance:
++
+[source,terminal]
+----
+oc apply -n <your-namespace> -f postgress-external-db-cluster-cert.yaml
+----
+
+. Configure your PostgreSQL instance in the Helm configuration file named `values.yaml`:
++
+[source,yaml]
+----
+# ...
+upstream:
+ postgresql:
+   enabled: false  # disable PostgreSQL instance creation (1)
+ backstage:
+   appConfig:
+     backend:
+       database:
+          connection:  # configure Backstage DB connection parameters
+            host: ${POSTGRES_HOST}
+            port: ${POSTGRES_PORT}
+            user: ${POSTGRES_USER}
+            password: ${POSTGRES_PASSWORD}
+            ssl:
+              rejectUnauthorized: true,
+              ca:
+                $file: /opt/app-root/src/postgres-ca.pem
+              key:
+                $file: /opt/app-root/src/postgres-key.key
+              cert:
+                $file: /opt/app-root/src/postgres-crt.pem
+         # ...
+    extraVolumeMounts:
+      # The initContainer below will install dynamic plugins in this volume mount.
+      - mountPath: /opt/app-root/src/postgres-crt.pem
+        name: postgress-external-db-cluster-cert # inject certificate secret to Backstage
+        subPath: tls.crt
+      - mountPath: /opt/app-root/src/postgres-ca.pem
+        name: postgress-external-db-cluster-cert # inject certificate secret to Backstage
+        subPath: ca.crt
+      - mountPath: /opt/app-root/src/postgres-key.key
+        name: postgress-external-db-cluster-cert # inject certificate secret to Backstage
+        subPath: tls.key
+    extraVolumes:
+      # ...
+      - name: postgress-external-db-cluster-cert
+        secret:
+          secretName: postgress-external-db-cluster-cert  <2>      
+        # ...
+    initContainers:
+      # ...
+    extraEnvVarsSecrets:
+      - postgres-cred # inject credential secret to Backstage <3>
+    auth:
+      existingSecret: postgres-cred      
+# ...    
```

Comment:
I wasn't able to deploy anything using this configuration. I believe it is better to use the values from modules/admin/proc-configuring-postgresql-instance-using-helm.adoc there

Reply:
I used the values from the mentioned adoc file but here we need an extra CA certificate `postgres-ca.pem` and a TLS private key `postgres-key.key`. So, i have kept those values.

 BTW, I earlier referred the values from this [PR](https://github.com/janus-idp/backstage-showcase/pull/1306/files#diff-ecc5adb08cc5187e34799ef00c41bd44fd4d90df8e513731707b30ff203d0ec3).